include/psa/crypto.h

```diff
@@ -4043,6 +4043,553 @@ psa_status_t psa_generate_random(uint8_t *output,
 psa_status_t psa_generate_key(const psa_key_attributes_t *attributes,
                               mbedtls_svc_key_id_t *key);
 
+/**@}*/
+
+/** \defgroup interruptible_hash Interruptible sign/verify hash
+ * @{
+ */
+
+/** The type of the state data structure for interruptible hash
+ *  verification operations.
+ *
+ * Before calling any function on a sign hash operation object, the
+ * application must initialize it by any of the following means:
+ * - Set the structure to all-bits-zero, for example:
+ *   \code
+ *   psa_sign_operation_t operation;
+ *   memset(&operation, 0, sizeof(operation));
+ *   \endcode
+ * - Initialize the structure to logical zero values, for example:
+ *   \code
+ *   psa_sign_operation_t operation = {0};
+ *   \endcode
+ * - Initialize the structure to the initializer
+ *   #PSA_SIGN_HASH_OPERATION_INIT, for example:
+ *   \code
+ *   psa_sign_operation_t operation = PSA_SIGN_HASH_OPERATION_INIT;
+ *   \endcode
+ * - Assign the result of the function psa_sign_hash_operation_init() to
+ *   the structure, for example:
+ *   \code
+ *   psa_sign_operation_t operation;
+ *   operation = psa_sign_hash_operation_init();
+ *   \endcode
+ *
+ * This is an implementation-defined \c struct. Applications should not
+ * make any assumptions about the content of this structure.
+ * Implementation details can change in future versions without notice. */
+typedef struct psa_sign_operation_s psa_sign_hash_operation_t;
```

Comment:
We're going to have a naming consistency issue with interruptible and multipart sign/verify-message. There are 8 combinations of signature operations: hash/message, single/multi-part, direct/interruptible. Multipart with hash is pointless since a hash is a very small input. If you need interruptibility with a message you probably need it multipart. On the other hand, multipart non-interruptible is useful to have in the API because interruptibility is a niche requirement. Operation objects are only used if the operation is multipart or interruptible. This leaves the following operation types:

* Hash, single-part, direct: no operation.
* Hash, single-part, interruptible: this.
* Message, single-part, direct: no operation.
* Message, multi-part, direct: ?
* Message, multi-part, interruptible: ?